controllers/machinedeployment_sync.go

```diff
@@ -413,7 +414,7 @@ func (r *MachineDeploymentReconciler) scaleMachineSetOperation(ms *clusterv1.Mac
 		*(ms.Spec.Replicas) = newScale
 		mdutil.SetReplicasAnnotations(ms, *(deployment.Spec.Replicas), *(deployment.Spec.Replicas)+mdutil.MaxSurge(*deployment))
 
-		err = r.Patch(context.Background(), ms, patch)
+		err = r.Client.Patch(context.Background(), ms, patch)
```

Comment:
Does this need to operate on a copy too?

Reply:
In this case, I don't think so, because we only update the Spec of the MachineSet from within the MachineDeployment controller.